marco-chelsea/public/scripts/article.js

```diff
@@ -0,0 +1,89 @@
+'use strict';
+
+function Article (rawDataObj) {
+  Object.keys(rawDataObj).forEach(function(key) {
+    this[key] = rawDataObj[key]
+  }, this);
+}
+
+Article.all = [];
+
+Article.prototype.toHtml = function() {
+  var template = Handlebars.compile($('#article-template').text());
+
+  this.daysAgo = parseInt((new Date() - new Date(this.publishedOn))/60/60/24/1000);
+  this.publishStatus = this.publishedOn ? `published ${this.daysAgo} days ago` : '(draft)';
+  this.body = marked(this.body);
+
+  return template(this);
+};
+
+Article.loadAll = function(rows) {
+  rows.sort(function(a,b) {
+    return (new Date(b.publishedOn)) - (new Date(a.publishedOn));
+  });
+
+  Article.all = rows.map(function(ele) {
+    return new Article(ele);
+  });
+};
+
+Article.fetchAll = function(callback) {
+  $.get('/articles')
+  .then(
+    function(results) {
+      Article.loadAll(results);
+      callback();
+    }
+  )
+};
+
+Article.truncateTable = function(callback) {
+  $.ajax({
+    url: '/articles',
+    method: 'DELETE',
+  })
+  .then(function(data) {
+    console.log(data);
+    if (callback) callback();
+  });
+};
+
+Article.prototype.insertRecord = function(callback) {
```

Comment:
Check for errors in the SQL query, insertRecord() is currently not functional. You can test this function by following the instructions here: https://github.com/codefellows-seattle-301d24/09-sql-joins-relations/blob/master/CRUD-testing.md